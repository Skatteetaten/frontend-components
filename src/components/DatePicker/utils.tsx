export const DEFAULTFORMATDATE = (date: Date | null | undefined): string => {
  return date
    ? Intl.DateTimeFormat('no', {
        year: 'numeric',
        month: '2-digit',
        day: 'numeric',
      }).format(date)
    : '';
};

export const DEFAULTPARSEDATEFROMSTRING = (date: string): Date | null => {
  try {
    if (date && date.length === 10) {
      const day: number = parseInt(date.substring(0, 2));
      const month: number = parseInt(date.substring(3, 5));
      const year: number = parseInt(date.substring(6, 10));
      const formattedDate: Date = new Date(Date.UTC(year, month - 1, day));
      return formattedDate;
    }
  } catch (error) {
    console.log(error);
  }
  return null;
};

export const monthsForLocale = (
  localeName = 'no',
  monthFormat: 'long' | 'short' = 'long'
) => {
  const format = new Intl.DateTimeFormat(localeName, { month: monthFormat })
    .format;
  const months: string[] = [];
  for (let month = 0; month < 12; month++) {
    months.push(format(new Date(Date.UTC(2020, month))));
  }
  return months;
};

export const weekdaysForLocale = (
  localeName = 'no',
  dayFormat: 'long' | 'short' = 'long'
) => {
  const format = new Intl.DateTimeFormat(localeName, { weekday: dayFormat })
    .format;
  const weekdays: string[] = [];
  for (let day = 0; day < 7; day++) {
    weekdays.push(format(new Date(Date.UTC(2020, 7, day + 2)))); // add 2 days to let the week start on Sunday
  }
  return weekdays;
};
