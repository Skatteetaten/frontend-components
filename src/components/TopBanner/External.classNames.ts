import { mergeStyleSets } from '@uifabric/merge-styles';
import { getTheme } from '@uifabric/styling';
import { FontSizes } from '..';
import { PaletteProps } from '..';

const separator = require('./assets/separator.png');
// @ts-ignore TODO
export function getClassNames(props): string {
  const { compact } = props;
  const palette = getTheme().palette as PaletteProps;
  // @ts-ignore TODO
  return mergeStyleSets({
    header: {
      selectors: {
        '&::after': {
          display: 'block',
          content: '""',
          width: '100%',
          height: compact ? '12px' : '20px',
          backgroundColor: '#fff',
          backgroundRepeat: 'no-repeat',
          backgroundPosition: 'center',
          backgroundSize: 'cover',
          backgroundImage: `url(${separator})`
        }
      }
    },
    headerMain: {
      width: 'auto',
      marginRight: 'auto',
      marginLeft: 'auto',
      marginTop: 'auto',
      height: 'auto',
      padding: compact ? '8px' : '16px 16px 0 16px',
      display: 'flex',
      //flexDirection: 'column',
      background: palette.skeColor.white,
      maxWidth: '1100px',
      selectors: {
        '@media (min-width: 640px)': {
          width: 'auto',
          minWidth: compact ? 'auto' : '500px',
          flexDirection: 'row'
        }
      }
    },
    contentWrapper: {
      flexGrow: 3,
      flexDirection: 'colum',
      textAlign: 'left',
      width: '100%',
      paddingLeft: '16px',
      selectors: {
        '@media (min-width: 480px)': {
          paddingLeft: '32px'
        }
      }
    },
    title: {
      marginTop: '4px',
      fontSize: FontSizes.large,
      selectors: {
        '@media (min-width: 640px)': {
          fontSize: FontSizes.xLarge
        }
      }
    },
    separator: {
      width: '100%',
      height: '20px',
      backgroundColor: '#fff',
      backgroundRepeat: 'no-repeat',
      backgroundPosition: 'center',
      backgroundSize: 'cover',
      backgroundImage: `url(${separator})`
    },
    linkButton: {
      displayName: 'SkeActionButton',
      marginTop: '-12px',
      marginLeft: '-8px',
      selectors: {
        '& .ms-Button-textContainer': {
          color: palette.skeColor.blue,
          fontSize: FontSizes.xSmall,
          fontWeight: 'bold'
        }
      }
    }
  });
}
