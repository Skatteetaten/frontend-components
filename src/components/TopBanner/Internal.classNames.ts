import { FontSizes, PaletteProps } from '..';
import { getTheme } from '@uifabric/styling';
import { mergeStyleSets } from '@uifabric/merge-styles';
// @ts-ignore TODO
export const getClassNames = function getClassNames(props) {
  const { compact } = props;
  const sidebarWidth = '362px';
  const palette = getTheme().palette as PaletteProps;
  return mergeStyleSets({
    header: {
      margin: '0',
      borderRadius: '0 0 0 0',
      display: 'flex',
      overflow: 'hidden',
      background: palette.skeColor.internalLight,
      color: palette.skeColor.white,
      alignItems: 'stretch',
      selectors: {
        '@media (min-width: 640px)': {
          height: compact ? '72px' : '120px'
        }
      }
    },
    headerActionButton: {
      displayName: 'Lenkeknapp',
      marginLeft: '8px',
      selectors: {
        ':hover': {
          backgroundColor: 'rgba(255,255,255,.1)'
        },
        '&& i, &&:hover i, &&:focus i, &&:active i': {
          color: palette.skeColor.white
        },
        '& .ms-Button-flexContainer': {
          color: palette.skeColor.white
        }
      }
    },
    headerLeftContainer: {
      // @ts-ignore TODO
      width: `${sidebarWidth - 40}px`,
      background: palette.skeColor.internal,
      display: 'block',
      alignItems: 'center',
      paddingLeft: 16,
      selectors: {
        '@media (min-width: 640px)': {
          display: 'flex'
        }
      }
    },
    headerLogo: {
      selectors: {
        img: {
          margin: '16px auto 0 auto',
          height: 50,
          selectors: {
            '@media (min-width: 640px)': {
              margin: 0,
              height: compact ? 50 : 90
            }
          }
        }
      }
    },
    headerDiagonal: {
      borderRight: `40px solid ${palette.skeColor.internalLight}`,
      borderBottom: `120px solid ${palette.skeColor.internal}`
    },
    headerRightContainer: {
      background: palette.skeColor.internalLight,
      color: palette.skeColor.black,
      padding: '0 8px',
      flexGrow: '1',
      display: 'flex',
      alignItems: 'center',
      wordBreak: 'break-word',
      selectors: {
        '& h1': {
          flexGrow: 1,
          fontSize: FontSizes.large,
          selectors: {
            '@media (min-width: 640px)': {
              fontSize: FontSizes.xLarge
            }
          }
        }
      }
    },
    headerRightButtons: {
      display: 'flex',
      alignSelf: 'flex-end'
    }
  });
};
