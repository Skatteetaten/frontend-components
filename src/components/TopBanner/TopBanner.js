import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Image from '../Image/Image';
import ActionButton from '../ActionButton/ActionButton';

import { getClassNames as getInternalClassNames } from './Internal.classNames';
import { getClassNames as getExternalClassNames } from './External.classNames';

const InternalHeader = props => {
  const styles = getInternalClassNames(props);
  const logo = require('./assets/ske-logo-intern.svg');
  return (
    <header
      className={classnames(styles.header, props.className)}
      id={props.id}
    >
      <div className={styles.headerLeftContainer}>
        <Image
          className={styles.headerLogo}
          src={logo}
          alt="Skatteetaten logo"
        />
        <div style={{ flexGrow: 1, textAlign: 'center' }}>
          <ActionButton
            href={props.homeUrl}
            className={styles.headerActionButton}
            icon="Home"
            color="white"
            ariaLabel="Til startsiden"
          >
            {props.homeText}
          </ActionButton>
        </div>
      </div>
      <div className={styles.headerDiagonal} />
      <div className={styles.headerRightContainer}>
        <h1>{props.title}</h1>
        <div className={styles.headerRightButtons}>{props.children}</div>
      </div>
    </header>
  );
};

const ExternalHeaderContent = ({ styles, ...props }) => {
  if (props.children) {
    return props.children;
  }

  if (props.compact) {
    return <h1 className={styles.title}>{props.title}</h1>;
  }

  return (
    <>
      <ActionButton
        href={props.onClick ? undefined : props.homeUrl}
        onClick={props.onClick}
        className={styles.linkButton}
        icon="Back"
        ariaLabel="Til startsiden"
      >
        {props.homeText}
      </ActionButton>
      <h1 className={styles.title}>{props.title}</h1>
    </>
  );
};

const ExternalHeader = props => {
  const styles = getExternalClassNames(props);
  const logo = require('./assets/ske-logo.svg');
  const compactHeight = props.compact ? 55 : 68;

  const imageElement = (
    <Image src={logo} height={compactHeight} alt="Skatteetaten logo" />
  );

  return (
    <header
      className={classnames(styles.header, props.className)}
      id={props.id}
    >
      <div className={styles.headerMain}>
        <div className={styles.logoWrapper}>
          <div>
            {props.logoLink ? (
              <a href="https://www.skatteetaten.no">{imageElement}</a>
            ) : (
              imageElement
            )}
          </div>
        </div>
        <div className={styles.contentWrapper}>
          <ExternalHeaderContent styles={styles} {...props} />
        </div>
      </div>
    </header>
  );
};

/**
 * @visibleName TopBanner (Topp)
 */
export default class TopBanner extends React.PureComponent {
  static propTypes = {
    /** Tittelen på banneren */
    title: PropTypes.string,
    /** Teksten som vises ved siden av home-knapp */
    homeText: PropTypes.string,
    /** URLen på homeknappen */
    homeUrl: PropTypes.string,
    /** Om banneren er ment for en intern eller ekstern løsning */
    external: PropTypes.bool,
    /** Om banneren skal ta mindre plass vertikalt  */
    compact: PropTypes.bool,
    /** Overstyring av stiler */
    className: PropTypes.string,
    /** Global attributt som må være unik for hele HTML dokumentet */
    id: PropTypes.string,
    /** Om logoen skal lenke til skatteetaten.no eller ikke (kun ekstern) */
    logoLink: PropTypes.bool
  };
  static defaultProps = {
    title: undefined,
    homeUrl: undefined,
    homeText: undefined,
    external: false,
    compact: false,
    logoLink: false
  };

  render() {
    const { external, ...props } = this.props;
    return external ? (
      <ExternalHeader {...props} />
    ) : (
      <InternalHeader {...props} />
    );
  }
}
