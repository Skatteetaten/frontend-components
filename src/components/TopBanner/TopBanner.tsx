import * as React from 'react';
import ActionButton from '../ActionButton/ActionButton';
import classnames from 'classnames';
import Image from '../Image/Image';
// @ts-ignore TODO
import logo from './assets/ske-logo-intern.svg';
import { getClassNames as getExternalClassNames } from './External.classNames';
import { getClassNames as getInternalClassNames } from './Internal.classNames';
// @ts-ignore TODO
const InternalHeader = props => {
  const styles = getInternalClassNames(props);
  return (
    <header
      className={classnames(styles.header, props.className)}
      id={props.id}
    >
      <div className={styles.headerLeftContainer}>
        <Image
          className={styles.headerLogo}
          src={logo}
          alt="Skatteetaten logo"
        />
        <div style={{ flexGrow: 1, textAlign: 'center' }}>
          <ActionButton
            href={props.homeUrl}
            className={styles.headerActionButton}
            icon="Home"
            color="white"
            ariaLabel="Til startsiden"
          >
            {props.homeText}
          </ActionButton>
        </div>
      </div>
      <div className={styles.headerDiagonal} />
      <div className={styles.headerRightContainer}>
        <h1>{props.title}</h1>
        <div className={styles.headerRightButtons}>{props.children}</div>
      </div>
    </header>
  );
};
// @ts-ignore TODO
const ExternalHeaderContent = ({ styles, ...props }) => {
  if (props.children) {
    return props.children;
  }

  if (props.compact) {
    return <h1 className={styles.title}>{props.title}</h1>;
  }

  return (
    <>
      <ActionButton
        href={props.onClick ? undefined : props.homeUrl}
        onClick={props.onClick}
        className={styles.linkButton}
        icon="Back"
        ariaLabel="Til startsiden"
      >
        {props.homeText}
      </ActionButton>
      <h1 className={styles.title}>{props.title}</h1>
    </>
  );
};
// @ts-ignore TODO
const ExternalHeader = props => {
  const styles = getExternalClassNames(props);
  // @ts-ignore TODO
  const { header, headerMain, logoWrapper, contentWrapper } = styles;
  const logo = require('./assets/ske-logo.svg');
  const compactHeight = props.compact ? 55 : 68;

  const imageElement = (
    <Image src={logo} height={compactHeight} alt="Skatteetaten logo" />
  );

  return (
    <header className={classnames(header, props.className)} id={props.id}>
      <div className={headerMain}>
        <div className={logoWrapper}>
          <div>
            {props.logoLink ? (
              <a href="https://www.skatteetaten.no">{imageElement}</a>
            ) : (
              imageElement
            )}
          </div>
        </div>
        <div className={contentWrapper}>
          <ExternalHeaderContent styles={styles} {...props} />
        </div>
      </div>
    </header>
  );
};

interface TopBannerProps {
  /** Tittelen på banneren */
  title?: string;
  /** Teksten som vises ved siden av home-knapp */
  homeText?: string;
  /** URLen på homeknappen */
  homeUrl?: string;
  /** Om banneren er ment for en intern eller ekstern løsning */
  external?: boolean;
  /** Om banneren skal ta mindre plass vertikalt  */
  compact?: boolean;
  /** Overstyring av stiler */
  className?: string;
  /** Global attributt som må være unik for hele HTML dokumentet */
  id?: string;
  /** Om logoen skal lenke til skatteetaten.no eller ikke (kun ekstern) */
  logoLink?: boolean;
}
/**
 * @visibleName TopBanner (Topp)
 */
export default class TopBanner extends React.PureComponent<TopBannerProps, {}> {
  static defaultProps = {
    title: undefined,
    homeUrl: undefined,
    homeText: undefined,
    external: false,
    compact: false,
    logoLink: false
  };

  render() {
    const { external, ...props } = this.props;
    return external ? (
      <ExternalHeader {...props} />
    ) : (
      <InternalHeader {...props} />
    );
  }
}
