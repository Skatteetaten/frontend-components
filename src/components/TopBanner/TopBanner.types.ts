import { ReactNode } from 'react';

export interface TopBannerTypes {
  /** Tittelen på banneren */
  title?: string | JSX.Element;
  /** Teksten som vises ved siden av home-knapp */
  homeText?: string;
  /** URLen på homeknappen */
  homeUrl?: string;
  /** Ikon på intern toppbanner */
  icon?: string;
  /** Om banneren er ment for en intern eller ekstern løsning */
  external?: boolean;
  /** Om banneren skal ta mindre plass vertikalt  */
  compact?: boolean;
  /** Overstyring av stiler */
  className?: string;
  /** Mulighet til å sette bredde på skrått område i toppen (kun intern) */
  slantedAreaWidth?: number | string;
  /** Global attributt som må være unik for hele HTML dokumentet */
  id?: string;
  /** Om logoen skal lenke til skatteetaten.no eller ikke (kun ekstern) */
  logoLink?: boolean;
  /** Lenken når man klikker på logoen */
  logoLinkUrl?: string;
  /** OnClick event som trigges av klikk på hjemlink */
  onClick?: () => void;
  /** Språk på logoen */
  language?: 'nb' | 'nn' | 'en' | 'se';
  /** Topstripe */
  topStripe?: JSX.Element;
  slantedAreaClassName?: string;
  children?: ReactNode;
}
