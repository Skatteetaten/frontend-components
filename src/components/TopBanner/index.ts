export { TopBanner as default } from './TopBanner';
export * from './TopBanner';
