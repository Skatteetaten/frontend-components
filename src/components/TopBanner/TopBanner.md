** Toppbanner vises på toppen i løsningene og skiller interne og eksterne løsninger fra hverandre **

### Publikumsløsninger

Ikke-innlogget publikumsløsning:

```js
import TopBanner from '@skatteetaten/frontend-components/TopBanner';

<TopBanner
  external
  homeText="Til skatteetaten.no"
  title="Ekstern publikumsløsning"
  logoLink
/>;
```

Innlogget publikumsløsning:

```js
import TopStripe, {
  TopStripeMenu,
  TopStripeButton
} from '@skatteetaten/frontend-components/TopStripe';
import TopBanner from '@skatteetaten/frontend-components/TopBanner';
import Link from '@skatteetaten/frontend-components/Link';
import Icon from '@skatteetaten/frontend-components/Icon';

<div>
  <TopStripe>
    <Link path={'#topbanner'} text={'Kontakt oss'} placement="before" />

    <TopStripeMenu title={'Endre skriftstørrelse'}>
      <div style={{ fontSize: '20px' }}>
        Hold Ctrl-tasten nede (Cmd-tasten på Mac). Trykk på + for å forstørre
        eller - for å forminske.
      </div>
    </TopStripeMenu>
    <TopStripeMenu title={'Language / Språk'}>
      <TopStripeButton ariaLabel={'Norsk'} onClick={() => console.log('NB')}>
        Norsk
      </TopStripeButton>
      <TopStripeButton
        icon={'check'}
        ariaLabel={'Nynorsk'}
        onClick={() => console.log('NN')}
      >
        Nynorsk
      </TopStripeButton>
      <TopStripeButton ariaLabel={'Engelsk'} onClick={() => console.log('EN')}>
        Engelsk
      </TopStripeButton>
      <TopStripeButton
        ariaLabel={'Sørsamisk'}
        onClick={() => console.log('SMA')}
      >
        Sørsamisk
      </TopStripeButton>
      <TopStripeButton
        ariaLabel={'Nordsamisk'}
        onClick={() => console.log('SME')}
      >
        Nordsamisk
      </TopStripeButton>
    </TopStripeMenu>

    <span>
      <Icon iconName="person" />
      Vegard Sandli
    </span>

    <Link path={'#topbanner'} text={'Logg ut'} placement="before" />
  </TopStripe>
  <TopBanner
    external
    title={'Side for publikum'}
    homeText={'Tilbake til skatteetaten.no'}
    logoLink
  />
</div>;
```

### Interne løsninger

På startsiden i et fagsystem:

```js
import TopBanner from '@skatteetaten/frontend-components/TopBanner';

<div>
  <TopBanner compact homeText="Systemnavn" homeUrl="#topbanner">
    <div></div>
  </TopBanner>
</div>;
```

På en underside i et fagsystem:

```js
import TopBanner from '@skatteetaten/frontend-components/TopBanner';

<div>
  <TopBanner
    compact
    homeText="Systemnavn"
    title="Sak eller arbeidsoppgave"
    homeUrl="#topbanner"
  >
    <div></div>
  </TopBanner>
</div>;
```

```js noeditor beskrivelse
<h3>Forskjell på interne systemer og publikumsløsninger</h3>
  <p>
   Alle løsningene våre har et toppbanner som visuelt skiller interne og eksterne fra hverandre. Interne løsninger bruker fylt toppbanner med skrå strek, mens eksterne bruker en smal lang strek, slik som på <a href='https://www.skatteetaten.no'>skatteetaten.no</a>. Dette lager en smidig overgang fra skatteetaten.no til en tilknyttet løsning.
  </p>
  <p>
    Det er valgfritt å bruke kompakt eller standard utgave av banneret, men du må være konsekvent gjennom hele løsningen.
  </p>
  <h3>Startsider på interne løsinger</h3>
  <p>Interne fagsystemer har gjerne en hovedside der brukerne finner arbeidsoppgaver eller søker etter opplysninger. Her kan toppbanneret holdes enkel, og kun inneholde navn på logo og navn på applikasjonen. Toppbanneret kan inneholde meny til rapporter og «løse sider», eller personlige innstillinger. Du trenger ikke skrive «Hjemmeside», «Startside», «Velkommen» i toppbanneret her.</p>
  <h3>Undersider på interne løsinger</h3>
   <p>De fleste interne systemer har en eller flere undersider. Banneret på sidene bør ha en tittel som forteller konteksten til innholdet på siden, altså en forklaring som gjør det relevant for brukeren, for eksempel en sakstype eller arbeidsoppgave. Dersom du velger å ha arbeidsoppgaven i toppbanneret, skal også arbeidsliste-funksjonene «Gjør tilgjengelig», «Sett på vent» og «Fordel» stå i umiddelbar nærhet til arbeidsoppgaven.
   </p>
    <h3>Klikkbar logo</h3>
  <p>
    For alle publikumsløsninger skal logoen som hovedregel være klikkbar og ta deg
    til forsiden av <a href='https://www.skatteetaten.no'>skatteetaten.no</a>.
  </p>

  <p>
    Se også {' '}
    <a href="https://www.skatteetaten.no/stilogtone/visuelt/logo/">
      visuell identitet
    </a>{' '}
    for detaljer rundt bruk av logoen.
  </p>
```

```js noeditor uu
<div>
  <h3>Tips</h3>
  <ul>
    <li>
      Valgene i toppstripa som utvider innhold, bør ha et pilikon, for visuell
      indikasjon av funksjonaliteten.{' '}
    </li>
    <li>
      Interaktivt innhold i utvida innhold skal kunne velges med tastatur.
    </li>
    <li>
      Når du lukker utvidet innhold, for eksempel ved escape eller lukk-knappen,
      skal tastaturfokus være tilbake på hovedvalget.
    </li>
  </ul>

  <h3>Mest relevante WCAG-krav</h3>
  <ul>
    <li>2.1.1 A, Tastatur</li>
    <li>2.4.1 A, Hoppe over blokker</li>
    <li>2.4.3 A, Fokusrekkefølge</li>
    <li>4.1.2 A, Navn, rolle, verdi</li>
  </ul>

  <h3>WAI-ARIA</h3>
  <ul>
    <li>
      Role=link brukes på Tilbake til skattetaten.no-knappen fordi den visuelt
      ser ut som en lenke.
    </li>
    <li>
      Aria-expanded brukes der innhold utvides/minimeres, f.eks. Endre
      skriftstørrelse.
    </li>
    <li>
      Aria-expanded, aria-haspopup, aria-controls, aria-labelledby, role=menu,
      role=menuitem brukes på f.eks. på språkmenyen. Dette får skjermleser til å
      «bytte» modus og trigge menynavigering.
    </li>
    <li>Aria-hidden brukes for å skjule ikoner for skjermleser.</li>
  </ul>
</div>
```
