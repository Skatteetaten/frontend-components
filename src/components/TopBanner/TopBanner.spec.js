import React from 'react';
import { mount } from 'enzyme';
import { matches } from '../utils/test-utils';
import TopBanner from './TopBanner';

function oppsettFullDOM(props) {
  return mount(<TopBanner {...props} />);
}
describe('TopBanner komponent', () => {
  it('matcher snapshot ', () => {
    matches(<TopBanner />);
  });

  it('rendrer TopBanner med deafult props', () => {
    const wrapper = oppsettFullDOM();

    expect(wrapper.prop('title')).toEqual(undefined);
    expect(wrapper.prop('homeUrl')).toEqual(undefined);
    expect(wrapper.prop('homeText')).toEqual(undefined);
    expect(wrapper.prop('external')).toEqual(false);
    expect(wrapper.prop('compact')).toEqual(false);
  });

  it('rendrer TopBanner med riktige props', () => {
    const wrapper = oppsettFullDOM({
      className: 'topbanner-classname',
      id: 'toppbanner-id'
    });

    expect(wrapper.prop('className')).toEqual('topbanner-classname');
    expect(wrapper.prop('id')).toEqual('toppbanner-id');
  });

  it('rendrer TopBanner med visning for eksterne arbeidsflater ', () => {
    const wrapper = oppsettFullDOM({
      external: true,
      homeText: 'Til skatteetaten.no',
      title: 'Ekstern publikumsløsning'
    });

    expect(wrapper.find('h1').html()).toContain('Ekstern publikumsløsning');
    expect(wrapper.html()).toContain('contentWrapper');
    expect(wrapper.html()).toContain('headerMain');
    expect(wrapper.find('.ms-Icon').prop('data-icon-name')).toEqual('Back');
    expect(wrapper.find('ImageBase').prop('height')).toEqual(68);
    expect(wrapper.find('img').prop('alt')).toEqual('Skatteetaten logo');
  });

  it('rendrer TopBanner med kompakt visning for eksterne arbeidsflater ', () => {
    const wrapper = oppsettFullDOM({
      external: true,
      compact: true
    });

    expect(wrapper.find('ImageBase').prop('height')).toEqual(55);
    expect(wrapper.find('StyledIconBase')).toHaveLength(0);
  });

  it('rendrer TopBanner med kompakt visning for interne arbeidsflater ', () => {
    const wrapper = oppsettFullDOM({
      external: false,
      compact: true,
      title: 'Intern løsning'
    });

    expect(wrapper.find('header'));
    expect(wrapper.find('h1').html()).toContain('Intern løsning');
    expect(wrapper.find('i').prop('data-icon-name')).toEqual('Home');
    expect(wrapper.html()).toContain('headerDiagonal');
    expect(wrapper.html()).toContain('headerRightContainer');
    expect(wrapper.find('img').prop('alt')).toEqual('Skatteetaten logo');
  });

  it('rendrer intern TopBanner med annet ikon', () => {
    const wrapper = oppsettFullDOM({
      external: false,
      icon: 'ArrowBack',
      title: 'Intern løsning'
    });

    expect(wrapper.find('header'));
    expect(wrapper.find('h1').html()).toContain('Intern løsning');
    expect(wrapper.find('i').prop('data-icon-name')).toEqual('ArrowBack');
    expect(wrapper.html()).toContain('headerDiagonal');
    expect(wrapper.html()).toContain('headerRightContainer');
    expect(wrapper.find('img').prop('alt')).toEqual('Skatteetaten logo');
  });

  it('rendrer intern TopBanner uten tittel', () => {
    const wrapper = oppsettFullDOM({
      external: false,
      title: 'Intern løsning'
    });

    expect(wrapper.find('header'));
    expect(wrapper.contains('h1')).toBe(false);
    expect(wrapper.find('i').prop('data-icon-name')).toEqual('Home');
    expect(wrapper.html()).toContain('headerDiagonal');
    expect(wrapper.html()).toContain('headerRightContainer');
    expect(wrapper.find('img').prop('alt')).toEqual('Skatteetaten logo');
  });
});
