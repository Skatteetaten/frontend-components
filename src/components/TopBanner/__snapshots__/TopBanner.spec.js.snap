// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBanner komponent matcher snapshot  1`] = `
<header
  className=

      {
        align-items: stretch;
        background: #ECE6E8;
        border-radius: 0 0 0 0;
        color: #ffffff;
        display: flex;
        height: 120px;
        margin-bottom: 0;
        margin-left: 0;
        margin-right: 0;
        margin-top: 0;
        overflow: hidden;
      }
      @media (max-width: 640px){& {
        height: 72px;
      }
>
  <div
    className=

        {
          align-items: center;
          background: #6F2C3F;
          display: flex;
          padding-left: 16px;
          padding-right: 8px;
          width: auto;
        }
  >
    <a
      className=

          {
            align-items: center;
            display: flex;
            font-size: 1.125rem;
            font-weight: 700;
            text-decoration-color: #ffffff;
            text-decoration: none;
          }
    >
      <i
        aria-hidden={true}
        className=

            {
              -moz-osx-font-smoothing: grayscale;
              -webkit-font-smoothing: antialiased;
              color: #ffffff;
              display: inline-block;
              font-family: "Material Icons";
              font-size: 1.25rem;
              font-style: normal;
              font-weight: 700;
              margin-left: 10px;
              speak: none;
            }
        data-icon-name="Home"
      >
        ï‹œ
      </i>
      <div
        className=

            {
              color: #ffffff;
              flex-grow: 1;
              margin-left: 4px;
              text-align: center;
            }
            &:hover {
              text-decoration: underline;
            }
            &:active {
              text-decoration: none;
            }
      />
    </a>
  </div>
  <div
    className=

        {
          border-bottom: 120px solid #6F2C3F;
          border-right: 40px solid #ECE6E8;
        }
  />
  <div
    className=

        {
          align-items: center;
          background: #ECE6E8;
          color: #000000;
          display: flex;
          flex-grow: 1;
          padding-bottom: 0;
          padding-left: 8px;
          padding-right: 8px;
          padding-top: 0;
          word-break: break-word;
        }
        & h1 {
          flex-grow: 1;
          font-size: 1.125rem;
        }
        @media (min-width: 640px){& h1 {
          font-size: 1.375rem;
        }
  >
    <div
      className=

          {
            flex-grow: 1;
            font-size: 1.125rem;
          }
    />
    <div />
  </div>
</header>
`;
